99-general-concepts/02-testing/02-unit-tests/README.md

```diff
@@ -0,0 +1,128 @@
+# Unit Testing
+
+- Tipo: `lectura`
+- Formato: `self-paced`
+- Duración: `30min`
+
+***
+
+## Objetivos de Aprendizaje
+
+- Understand how to write effective unit tests
+- Understand prinicples behind Test Driven Development
+
+***
+
+## Introduction
+
+By now you’ve noticed that at the end of every code challenge you’ve had to run tests against your code to verify that it works as expected. These are called unit tests. Each test programmatically ensures that your program works as intended by comparing actual outputs with expected values for static inputs.
+
+For example, consider a function that adds to numbers: `sum(a, b)`.
+
+To test `sum(a,b)`, we can select any two numbers and assert that the ouptut equals their expected sum:
+
+```javascript
+Assert.deepEqual(sum(2, 3), 5);
+```
+
+If the result is false, meaning `sum(2, 3)` equals something other than 5, we'll know we have some debugging to do!
+
+## Validating Credentials
+
+To explore unit tests further, let's analyize the `areValidCredentials` challenge:
+
+>Escribe una función llamada `areValidCredentials`.
+>
+>Dando un nombre y una contraseña, `areValidCredentials`, devuelve `true` si el nombre es superior a 3 caracteres, Y, la contraseña tiene al menos 8 caracteres de longitud. De lo contrario, devuelve `false`.
+
+Let's look at the unit tests associated with this challenge:
+
+```javascript
+describe("areValidCredentials()", function () {
```

Comment:
Space between `function` and `()`. Since we use our style guide, it should be:

`function() { ... }`